### PR TITLE
Add insfrastructure folder to project template

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+ Add `infrastructure` folder to project template
+ Flag --no-infrastructure to create project with `infrastructure`
+ Update test
+ Update `guide` and `CHANGELOG.md`

### CODE REVIEW

File: docs/guide/README.md [hunk 1]
```diff
@@ -136,14 +136,21 @@ This will create a directory `my-project` and put multiple files into it.
...
+You can initial project without `infrastructure\` folder by use command:
+
+```bash
+algob init my-project --no-infrastructure
+```
+
 A `sample-project` is provided for your reference.
 
-Further information about the `sample-project` can be found [here](https://github.com/scale-it/algo-builder/blob/master/packages/algob/sample-project/README.md)
+Further information about the `sample-project` can be found [here](https://github.com/scale-it/algo-builder/blob/master/packages/algob/sample-project/js/README.md)
```
Comment (robert-zaremba) [inline docs/guide/README.md:153]:
```suggestion
Further information about the `sample-project` can be found in sample project [README](https://github.com/scale-it/algo-builder/blob/master/packages/algob/sample-project/js/README.md) file.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,11 @@ Features, Bug Fixes, Breaking Changes, Deprecated
 
 - Used app account instead of `deposit_lsig` in `examples/dao`
 - Support RekeyTo field in the inner transaction for TEAL v6.
+- Default add `infrastructure/` folder when initial algob project. You can create algob project without `infrastructure/` by use `--no-infrastructure` flag.  Example:
+
+```base
+  algob init --no-infrastructure
+```
```
Comment (robert-zaremba) [inline CHANGELOG.md:26]: I'm really wondering what should be the default... @ratik21  did you think more about it?

**Review by robert-zaremba — COMMENTED** [review]
looks good, left few language updates.

File: CHANGELOG.md (hunk 2 above)
Comment (vuvth) [reply CHANGELOG.md:26]: > I'm really wondering what should be the default... @ratik21 did you think more about it?

What happened if developer use the third party API? We don't need `infrastructure` in this case. Developer only want change API key in algob.config.js file to private network, test net, main net. I think add infrastructure should be optional. How to think about it @ratik21 @robert-zaremba ?

Comment (ratik21) [reply CHANGELOG.md:26]: Right, i think it's OK to assume developer will move to testnet/mainnet. But probably everyone starts project development on a dev network (it's very fast).. So i think `infrastructure` should be copied by default

File: docs/guide/README.md [hunk 3]
```diff
@@ -136,14 +136,21 @@ This will create a directory `my-project` and put multiple files into it.
 
 In the `my-project` folder you'll have following items:
 
+- `infrastructure/`: Directory for initial infrastructure commands. For more details please read `infrastructure/README.md`.
 - `assets/`: Directory for assets and contracts files
 - `scripts/`: Directory for scripts to deploy and run your assets and contracts
 - `tests/`: Directory for test files for testing your assets and contracts
 - `algob.config.js`: Algob configuration file
 
+You can initial project without `infrastructure\` folder by use command:
```
[the remark this answers is not in the document]
Comment (ratik21) [reply docs/guide/README.md:145]: `initialize project`

**Review by ratik21 — APPROVED** [review]
lgtm

File: CHANGELOG.md (hunk 2 above)
Comment (ratik21) [reply CHANGELOG.md:26]: Right, i think it's safe to assume user will move to testnet/mainnet. But i think probably everyone starts with dev network for development (it's very fast)..An alternative could be using api-as-a-service (purestake). But i think we should copy the `infrastructure` dir by default

File: docs/guide/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply docs/guide/README.md:145]: updated

File: packages/algob/src/internal/cli/project-creation.ts [hunk 4]
```diff
@@ -54,6 +54,11 @@ function copySampleProject (location: string, isTSProject: boolean): void {
   // copy common files first
   copy(path.join(sampleProjDir, "common"), location);
 
+  // copy infrastructure folder depending of --no-infrastructure flag
+  if (isUseInfrastucture) {
```
Comment (robert-zaremba) [inline packages/algob/src/internal/cli/project-creation.ts:58]:
```suggestion
  if (withInfrastucture) {
```

File: packages/algob/src/internal/cli/project-creation.ts [hunk 5]
```diff
@@ -87,10 +92,12 @@ async function printPluginInstallationInstructions (): Promise<void> {
   console.log(`  ${cmd.join(" ")}`);
 }
 
-export async function createProject (location: string, isTSProject: boolean): PromiseAny {
+export async function createProject (
+  location: string, isTSProject: boolean, isUseInfrastructure: boolean
```
Comment (robert-zaremba) [inline packages/algob/src/internal/cli/project-creation.ts:96]:
```suggestion
  location: string, isTSProject: boolean, withInfrastucture: boolean
```

**Review by robert-zaremba — APPROVED** [review]
Left few suggestions.

**Pull request merged**